MHArmory.Search/OpenCL/SearchResultDeserializer.cs

```diff
@@ -7,52 +7,60 @@
+                .Concat(data.AllWaists)
+                .Concat(data.AllLegs)
+                .Concat(data.AllCharms)
+                .Select(x => x.Equipment)
+                .ToDictionary(x => (x.Id, x.Type));
             var decoDict = data.AllJewels.ToDictionary(x => x.Jewel.Id);
 
             var results = new List<ArmorSetSearchResult>();
 
             var ms = new MemoryStream(serializedResults.Data);
             var reader = new BinaryReader(ms);
 
-            for (int i = 0; i < serializedResults.ResultCount; ++i)
+            for (int i = 0; i < serializedResults.ResultCount; i++)
             {
                 var result = new ArmorSetSearchResult();
                 result.ArmorPieces = new List<IArmorPiece>();
                 result.Jewels = new List<ArmorSetJewelResult>();
                 result.IsMatch = true;
-
-                for (int j = 0; j < 5; ++j)
+                const int equipmentTypesWithoutCharm = SearchLimits.EquipmentTypes - 1;
+                for (int j = 0; j < equipmentTypesWithoutCharm; j++)
                 {
                     ushort id = reader.ReadUInt16();
-                    var tuple = new Tuple<int, EquipmentType>(id, (EquipmentType) j + 1);
-                    var armor = (IArmorPiece)equipmentDict[tuple];
+                    var armor = (IArmorPiece)equipmentDict[(id, (EquipmentType)j + 1)];
                     result.ArmorPieces.Add(armor);
                 }
 
                 ushort charmId = reader.ReadUInt16();
-                var charmTuple = new Tuple<int, EquipmentType>(charmId, EquipmentType.Charm);
-                result.Charm = (ICharmLevel)equipmentDict[charmTuple];
+                result.Charm = (ICharmLevel)equipmentDict[(charmId, EquipmentType.Charm)];
 
                 byte decorationCount = reader.ReadByte();
-                for (int j = 0; j < decorationCount; ++j)
+                for (int j = 0; j < decorationCount; j++)
                 {
                     ushort id = reader.ReadUInt16();
                     byte count = reader.ReadByte();
                     IJewel jewel = decoDict[id].Jewel;
-                    var jewelResult = new ArmorSetJewelResult();
-                    jewelResult.Jewel = jewel;
-                    jewelResult.Count = count;
+                    var jewelResult = new ArmorSetJewelResult
+                    {
+                        Jewel = jewel,
+                        Count = count
+                    };
                     result.Jewels.Add(jewelResult);
                 }
 
-                int remainingDecos = 7 * 3 - decorationCount;
-                reader.ReadBytes(remainingDecos * 3);
-                result.SpareSlots = new int[3];
+                int remainingDecos = LengthConstants.DecorationsPerResult - decorationCount;
+                reader.ReadBytes(remainingDecos * LengthConstants.DecorationLength);
+                reader.ReadByte(); // Read dummy byte for 0-sized slots
```

Comment:
Note that here you can use `Seek(1, SeekOrigin.Current)` but for a single byte, no sure which one is faster.
Just to let you know in case you need to skip large chunks, seek is very useful and works the same as `fseek` in C/C++.